### ISSUE TITLE
Starting single node of an existing cluster fails

### ISSUE THREAD

**Issue by matthewGEA** [emqx/emqx#4535]
**Environment**:

- EMQ X version (e.g. `emqx_ctl status`):  4.3-rc.3 (docker pull emqx/emqx:4.3-rc.3)
- Hardware configuration (e.g. `lscpu`): x86_64, 4-core 16GB ram
- OS (e.g. `cat /etc/os-release`): ubuntu server 18.04
- Kernel (e.g. `uname -a`): 4.15.0-121-generic #123-Ubuntu SMP Mon Oct 5 16:16:40 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
- Erlang/OTP version (in case you build emqx from source code): N/A
- Others:

**How to reproduce it (as minimally and precisely as possible)**:
1. Connect two nodes in a cluster using mcast
2. power off both nodes
3. bring up one node
4. evaluate if it starts quickly.  If if starts quickly, power down, then power up the other node.
5. Observe that emqx waits about 2.5minutes at the "Starting emqx on node blah.localdomain" log message, then it will start.  After starting it outputs errors and does not start the dashboard or rules engine
6. Observe that starting the "fast node" allows the log errors to stop, but does not recover from the failed plugins on start (see the log snippet below)

**What happened and what you expected to happen**:
The node that was powered off after the other node is able to start emqx quickly.  If this "fast node" is powered off, then the other node (slow node) does not start correctly when powered up alone.  It can only be resolved by restarting the "slow node" after the "fast node" has already booted.  There is no reference to power-up ordering per the Getting Started documentation: https://docs.emqx.io/en/broker/v4.3/getting-started/cluster.html#node-discovery-and-autocluster

**Anything else we need to know?**:
Seems to be related to similar these closed issues?
https://github.com/emqx/emqx/issues/694
https://github.com/emqx/emqx/issues/3923

Snippet of log from docker container.  The "[error] [Stats] Update rule_registery_stats failed: function_clause" occurs ever 2 seconds:
```
Exec: /opt/emqx/erts-11.1.8/bin/erlexec -noshell -noinput +Bd -boot /opt/emqx/releases/4.3-rc.3/start -mode embedded -boot_var ERTS_LIB_DIR /opt/emqx/erts-11.1.8/../lib -mnesia dir "/opt/emqx/data/mnesia/blah.localdomain" -config /opt/emqx/data/configs/app.2021.04.12.03.18.06.config -args_file /opt/emqx/data/configs/vm.2021.04.12.03.18.06.args -vm_args /opt/emqx/data/configs/vm.2021.04.12.03.18.06.args -start_epmd false -epmd_module ekka_epmd -proto_dist ekka -- foreground
Root: /opt/emqx
Starting emqx on node blah.localdomain
2021-04-12 03:20:42.341 [error] Mnesia wait_for_tables timeout: [emqx_resource,mqtt_app,emqx_rule,emqx_banned,
                                 mqtt_admin]
Start http:management listener on 8081 successfully.
2021-04-12 03:20:42.500 [error] [Plugins] Load plugin emqx_dashboard failed, cannot start plugin emqx_dashboard for {bad_return,{{emqx_dashboard_app,start,[normal,[]]},{'EXIT',{{badmatch,{error,{shutdown,{failed_to_start_child,emqx_dashboard_admin,{aborted,{no_exists,[mqtt_admin,<<"admin">>]}}}}}},[{emqx_dashboard_app,start,2,[{file,"emqx_dashboard_app.erl"},{line,28}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,277}]}]}}}}

2021-04-12 03:20:42.573 [error] [Plugins] Load plugin emqx_rule_engine failed, cannot start plugin emqx_rule_engine for {bad_return,{{emqx_rule_engine_app,start,[normal,[]]},{'EXIT',{badarg,[{ets,match_object,[emqx_resource,'_'],[]},{ets,tab2list,1,[{file,"ets.erl"},{line,765}]},{emqx_rule_engine,refresh_resources,0,[{file,"emqx_rule_engine.erl"},{line,384}]},{emqx_rule_engine_app,start,2,[{file,"emqx_rule_engine_app.erl"},{line,31}]},{application_master,start_it_old,4,[{file,"application_master.erl"},{line,277}]}]}}}}
Start mqtt:tcp:internal listener on 127.0.0.1:11883 successfully.
Start mqtt:tcp:external listener on 0.0.0.0:1883 successfully.
Start mqtt:ws:external listener on 0.0.0.0:8083 successfully.
Start mqtt:ssl:external listener on 0.0.0.0:8883 successfully.
Start mqtt:wss:external listener on 0.0.0.0:8084 successfully.
EMQ X Broker 4.3-rc.3 is running now!
2021-04-12 03:20:44.348 [error] [Stats] Update rule_registery_stats failed: function_clause
2021-04-12 03:20:46.350 [error] [Stats] Update rule_registery_stats failed: function_clause
2021-04-12 03:20:48.352 [error] [Stats] Update rule_registery_stats failed: function_clause
2021-04-12 03:20:50.354 [error] [Stats] Update rule_registery_stats failed: function_clause
2021-04-12 03:20:52.356 [error] [Stats] Update rule_registery_stats failed: function_clause
2021-04-12 03:20:54.358 [error] [Stats] Update rule_registery_stats failed: function_clause


```

**Comment by qzhuyan** [comment]
trouble reproduced and solution is provided in 
https://github.com/emqx/ekka/pull/86

**Comment by matthewGEA** [comment]
@zmstone Can you see if version 4.2.11 grabbed this fix?  It doesn't seem to be working, but based on this fix being deployed 15 days ago and 4.2.11 was tagged 12 days ago, I would assume it got it?

**Comment by qzhuyan** [comment]
> @zmstone Can you see if version 4.2.11 grabbed this fix?  It doesn't seem to be working, but based on this fix being deployed 15 days ago and 4.2.11 was tagged 12 days ago, I would assume it got it?

Fix is on the master branch for 4.3 releases. It is included in 4.3 rc5.  will see if we can backport it to 4.2

**Comment by zmstone** [comment]
will be included in 4.2.12
https://github.com/emqx/emqx/pull/4725
